File: src/main/kotlin/media_item/implementations/Playlist.kt
package media_item.implementations
import media_item.MediaItem
import media_item_implementation.MediaItemImplementation
import media_item_implementation.implementations.PlaylistImplementation

class Playlist(
        override val title: String,
        override val creationDate: String,
        val description: String,
        val user: String,
        override val implementation: PlaylistImplementation
) : MediaItem() {
    val items = mutableListOf<MediaItem>()

    override fun displayInfo() {
        println("Playlist title: $title created by $user")
        println("Description: $description")
        println("Created on: $creationDate")
        println("Items:")
        items.forEach {
            println(it.title)
        }
    }

    override fun toString(): String {
        return "Playlist(title='$title', creationDate='$creationDate', description='$description', user='$user')"
    }
}

File: lab-1/domain/playlist.py
from domain.media_item import media_item


class Playlist(media_item):
    def __init__(self, title, creation_date, description, user):
        super().__init__(title, creation_date)
        self.description = description
        self.user = user

    def play(self):
        print(f"Playlist {self.title} is currently playing...")

    def display_info(self):
        print(f"Playlist title: {self.title} created by {self.user}")
        print(f"Description: {self.description}")
        print(f"Created on: {self.creation_date}")
